src/test/get_mediapipe_graph_metadata_response_test.cpp

```diff
@@ -107,3 +122,207 @@ TEST_F(GetMediapipeGraphMetadataResponse, BasicResponseMetadata) {
     EXPECT_EQ(thirdOutput.datatype(), "INVALID");
     EXPECT_EQ(thirdOutput.shape_size(), 0);
 }
+
+class MediapipeGraphDefinitionMetadataResponseBuild : public ::testing::Test {
+protected:
+    class MockMediapipeGraphDefinitionGetInputsOutputsInfo : public ovms::MediapipeGraphDefinition {
+        ovms::Status status = ovms::StatusCode::OK;
+
+    public:
+        MockMediapipeGraphDefinitionGetInputsOutputsInfo() :
+            MediapipeGraphDefinition("mediaDummy", {}, {}) {
+            MediapipeGraphDefinition::status.handle(ovms::ValidationPassedEvent());
+        }
+
+        void mockMetadata() {
+            // This functionality may be required in future graph tags implementations
+            std::string testPbtxt = R"(
+                input_stream: "TEST:in"
+                output_stream: "TEST0:out"
+                    node {
+                    calculator: "OVMSOVCalculator"
+                    input_stream: "B:in"
+                    output_stream: "A:out"
+                    }
+                )";
+        }
+
+        void mockStatus(ovms::Status status) {
+            this->status = status;
+        }
+
+        ovms::PipelineDefinitionStatus& getGraphDefinitionStatus() {
+            return ovms::MediapipeGraphDefinition::status;
+        }
+    };
+
+    MockMediapipeGraphDefinitionGetInputsOutputsInfo graphDefinition;
+    KFSModelMetadataResponse response;
+    ConstructorEnabledModelManager manager;
+
+public:
+    void prepare() {
+        graphDefinition.mockMetadata();
+    }
+};
+
+TEST_F(MediapipeGraphDefinitionMetadataResponseBuild, ModelVersionNotLoadedAnymoreButGraphNotReloadedYet) {
+    graphDefinition.mockStatus(ovms::StatusCode::MODEL_VERSION_NOT_LOADED_ANYMORE);
+    EXPECT_EQ(ovms::KFSInferenceServiceImpl::buildResponse(graphDefinition, &response), ovms::StatusCode::OK);
+}
+
+TEST_F(MediapipeGraphDefinitionMetadataResponseBuild, ModelVersionNotLoadedYet) {
```

Comment:
MP doesn't use at all information about model used inside to produce it's metadata so we don't need that test.